auth/store.go

```diff
@@ -129,6 +130,9 @@ type AuthStore interface {
 	// (The revision number is used for preventing the TOCTOU problem)
 	AuthInfoFromToken(token string) (*AuthInfo, bool)
 
+	// DeleteAuthInfo will delete the token once the connection using this token is closed
+	DeleteAuthInfo(token, remoteaddr string)
```

Comment:
I don't think the `auth` package should know anything about connections / remote addresses. It's poor software cohesion to introduce networking resources here.